tests/test.py

```diff
@@ -3823,6 +3842,155 @@ def testEmptyValues(self):
         self.assertEqual(row['binary'], b'')
 
 
+class TestOutdatedServiceConfigs(BaseTestCase, FailedRunProcessMixin):
+    def setUp(self):
+        return
+
+    def tearDown(self):
+        return
+
+    def testStartupWithoutVersionInConfig(self):
+        files = os.listdir('cmd/')
+        services = [i for i in files if os.path.isdir(os.path.join('cmd', i))]
+        self.assertTrue(services)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # generate configs for tests
+            subprocess.check_output(['configs/regenerate.sh', tmp_dir])
```

Comment:
Why do we regenerate configurations every time instead of keeping test input under version control?

I think it would be more simple to just keep a dozen of files ready to use, they will not suddenly change and lead to brittle tests. And if the tests _do_ fail one day then this indicates a breaking change in configuration format.

Reply:
why:
1. to test this script too
2. to not maintain these pre-generated files. if we add new binary in the future, we will need update `configs/regenerate.sh`, then update all test configs. now here we test 3 cases with different types of changes in configs. It's 3 folders with configs that we will need to remember what difference between these configs and regenerate them. Or add +3 scripts for each of these folders that will regenerate configs with all requirements for tests. And don't forget to remove these folders if we will remove any of test-cases.

in general, for me, it was more simple to call the script from test script instead pre-generate test files